src/entities/model.ts

```diff
@@ -14,7 +15,24 @@ export { ModelComponent, ModelTaskMetadata } from "./modelTaskMetadata";
  * A model observer: a function that takes a change set as input
  * and produces a list of tasks to process as output.
  */
-type ModelObserver = (changeBuffer: Immutable.Set<ModelComponent>) => Array<ModelTask>;
+export class ModelObserver {
+
+    private static counter: number = 0;
+    private identifier: number;
+
+    public constructor(private readonly observer: (changeBuffer: Immutable.Set<ModelComponent>) => Array<ModelTask>) {
+        this.identifier =
+        ModelObserver.counter++;
+    }
+
+    public getID(): number {
+        return this.identifier;
+    }
+
+    public getObserver(): (changeBuffer: Immutable.Set<ModelComponent>) => Array<ModelTask> {
```

Comment:
Have you considered changing this method's signature to `public observe(changeBuffer: Immutable.Set<ModelComponent>): Array<ModelTask>`? That's probably a lot more idiomatic in TypeScript. It also avoids the awkward `getObserver()(buffer)` call chain.